File: library/src/main/java/com/shxhzhxx/sdk/ui/LinearSpacingItemDecoration.java
package com.shxhzhxx.sdk.ui;

import android.graphics.Rect;
import android.view.View;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

public class LinearSpacingItemDecoration extends RecyclerView.ItemDecoration {
    public static final int HORIZONTAL = RecyclerView.HORIZONTAL;
    public static final int VERTICAL = RecyclerView.VERTICAL;

    private int space;
    private int orientation;
    private boolean includeEdge;

    public LinearSpacingItemDecoration(int space, int orientation, boolean includeEdge) {
        this.space = space;
        this.orientation = orientation;
        this.includeEdge = includeEdge;
    }

    @Override
    public void getItemOffsets(@NonNull Rect outRect, @NonNull View view, @NonNull RecyclerView parent, @NonNull RecyclerView.State state) {
        RecyclerView.Adapter adapter = parent.getAdapter();
        if (adapter == null)
            return;
        int position = parent.getChildAdapterPosition(view);
        int start = includeEdge || position > 0 ? space : 0;
        int end = includeEdge && position == adapter.getItemCount() - 1 ? space : 0;

        if (orientation == HORIZONTAL) {
            outRect.top = 0;
            outRect.bottom = 0;
            outRect.left = start;
            outRect.right = end;
        } else {
            outRect.left = 0;
            outRect.right = 0;
            outRect.top = start;
            outRect.bottom = end;
        }
    }
}

File: library/src/main/java/com/shxhzhxx/sdk/ui/LinearSpacingItemDecoration.kt
package com.shxhzhxx.sdk.ui

import android.graphics.Rect
import android.view.View
import androidx.recyclerview.widget.RecyclerView

class LinearSpacingItemDecoration(private val space: Int, @RecyclerView.Orientation private val orientation: Int, private val includeEdge: Boolean) : RecyclerView.ItemDecoration() {

    override fun getItemOffsets(outRect: Rect, view: View, parent: RecyclerView, state: RecyclerView.State) {
        val adapter = parent.adapter ?: return
        val position = parent.getChildAdapterPosition(view)
        val start = if (includeEdge || position > 0) space else 0
        val end = if (includeEdge && position == adapter.itemCount - 1) space else 0

        if (orientation == RecyclerView.HORIZONTAL) {
            outRect.top = 0
            outRect.bottom = 0
            outRect.left = start
            outRect.right = end
        } else {
            outRect.left = 0
            outRect.right = 0
            outRect.top = start
            outRect.bottom = end
        }
    }
}
